django/core/management/__init__.py

```diff
@@ -312,8 +312,11 @@ def execute(self):
                     autoreload.check_errors(django.setup)()
                 except Exception:
                     # The exception will be raised later in the child process
-                    # started by the autoreloader.
-                    pass
+                    # started by the autoreloader. Pretend it didn't happen by
+                    # loading an empty list of applications.
+                    apps.all_models = defaultdict(OrderedDict)
+                    apps.app_configs = OrderedDict()
+                    apps.apps_ready = apps.models_ready = apps.ready = True
```

Comment:
That seems convincing, especially if we'll throw out this code sometime soon anyway.